### ISSUE TITLE
Bank Journal - Auxiliary account - "NotDefined" is returned instead of a clear value

### ISSUE THREAD

**Issue by thibdrev** [Dolibarr/dolibarr#22744]
### Bug

In the Auxiliary Account column of the Bank Journal, `NotDefined` is returned instead of a default value:
![image](https://user-images.githubusercontent.com/7697736/199775335-16f20e2f-cc0c-4d92-9ff6-5303e9d273fd.png)

### Environment Version

16.0.1

### Environment OS

Linux builder.o2switch.net 4.18.0-348.12.2.lve.el7h.x86_64

### Environment Web server

Apache

### Environment PHP

7.4.30

### Environment Database

MySQL or MariaDB 5.5.5-10.5.17-MariaDB

### Environment URL(s)

/htdocs/accountancy/journal/bankjournal.php?id_journal=3

### Expected and actual behavior

* Expected behavior: 
    * A clear message such as: "Compte auxiliaire non défini ou tiers ou utilisateur inconnu. Nous utiliserons **421USER**"
* Actual behavior:
    * An unclear message: "Compte auxiliaire non défini ou tiers ou utilisateur inconnu. Nous utiliserons NotDefined"

### Steps to reproduce the behavior

At first I thought this was linked to the user accountancy account not being defined before I made the payment.
But I checked the following on a test instance: create a user and fill in its accountancy account > create an expense report, validate it and pay it. And I still get this behaviour.

The translated message is located here on Transifex:
https://www.transifex.com/dolibarr-association/dolibarr/translate/#fr_FR/accountancy/165994858
`Compte auxiliaire non défini ou tiers ou utilisateur inconnu. Nous utiliserons %s`

It is linked to the translation key `ThirdpartyAccountNotDefinedOrThirdPartyUnknown` which is used here:
https://github.com/Dolibarr/dolibarr/blob/57feb24f4acd82c25fe28c8706d902335d70220f/htdocs/accountancy/journal/bankjournal.php#L1287

So I'm guessing the `$tabcompany[$key]['code_compta']` is the `NotDefined` value.

**Comment by ouitec** [comment]
Hello,

Exactly in the same situation. We've upgraded to 16.0.3 and we still have the issue, when do you think you can fix this issue ?

We also have this issue on the modules "Paiements/Salaires" :

<img width="1355" alt="Capture d’écran 2023-01-25 à 17 17 30" src="https://user-images.githubusercontent.com/116264025/214630412-4a25d4ae-34ec-4738-bb61-1fa2e5b01343.png">

And "Paiements/Paiements Divers" : 

<img width="1356" alt="Capture d’écran 2023-01-25 à 18 05 31" src="https://user-images.githubusercontent.com/116264025/214631536-1342a25c-2348-40c7-96ab-23c0723fd280.png">

It was working before upgrade to 16.0.x I guess.

**Comment by Amael-PE** [comment]
I can report the same here, on a Doli `16.0.3`
I am investigating root causes in the code, without much success

**Comment by Amael-PE** [comment]
After a bit of investiguation, this is a false alert. The auxiliary code, if defined, is used. The point is taht it make impossible to know if it is defined...

**Comment by AurelienBISOTTI** [comment]
Then it is not a bug, but you might open a separate request if you think the message needs to change

**Comment by Amael-PE** [comment]
@AurelienBISOTTI I do think it is a bug
The code is defined, and the user alterted that it is not

**Comment by Amael-PE** [comment]
Unfortunately, even looking for a while to the code, I was not able to get where the issue comes from

**Comment by AurelienBISOTTI** [comment]
Then I don't get the "The auxiliary code, if defined, is used"
Where is it used ? In the bookeeping ? But not showed in bank account ?

More screenshots would be welcome, so I could test if there is still this issue in v18

**Comment by Amael-PE** [comment]
Yes, exactly. The auxiliary code is difned, but not showned in the Bank journal "ventilation", and used properly in the bookeeping

**Comment by Amael-PE** [comment]
In the "ventilation" for bank journal:
![Doli capture ventilation](https://github.com/Dolibarr/dolibarr/assets/41053833/87dad549-20a2-4b2f-a2fd-02d98ddfe929)

In the bookeeping, the exact same entry :
![Doli capture livre](https://github.com/Dolibarr/dolibarr/assets/41053833/a8cd2c47-23c6-4256-9f4e-461571d5167f)

**Comment by Amael-PE** [comment]
In the bookkeeping, everything is ok

**Comment by AurelienBISOTTI** [comment]
In bank journal, account is 220, but is 2310 in bookkeeping.

Are you sure bookkeeping was not modified afterward ?

Can you try again to create salary, and check it ?
From the bank journal, you can also check that this line is "in accouting"

![Screenshot-7](https://github.com/Dolibarr/dolibarr/assets/12530585/fab8b67d-58cf-4855-8da8-39ef5ed4eb7a)
![Capture d’écran du 2023-06-10 16-42-38](https://github.com/Dolibarr/dolibarr/assets/12530585/53533f9c-7d58-40f8-b98e-04f877e871d5)

**Comment by Amael-PE** [comment]
Yes, I am 200 % sure.
I did modified the **master** code in the bookeeping, because of another strange behaviour in Dolibarr, but I can guarantee that in `17.0.0`, use in prod, I still have the problem of missing **auxiliary count**. Thanks god, the auxiliary is working properly n the bookeeping ^^"
I create about 20 salaries per months, and then ventilate every beginning of following month, so I can ensure the bug is not fixed.
In the "Ventilation" part, the php doesn't get the auxiliary account for an unknown reason, and report it as missing, while it is not.

**Comment by AurelienBISOTTI** [comment]
How can you explain that "main" account number (not the auxiliary) has changed between the "ventilation" for bank journal and bookkeeping (220 vs 2310) ?

**Comment by Amael-PE** [comment]
I simply changed it inthe bookeeping, but this is a complete other story, definitly unrelated to the fact that:
 - `htdocs/accountancy/journal/bankjournal.php?mainmenu=accountancy&leftmenu=accountancy_journal&id_journal=3`
 - doesn't recognize auxiliary accounts related to users, thus printing an error to the user that proceed the validation/ventilation
 - while the bookeeping `htdocs/accountancy/bookkeeping/listbyaccount.php?mainmenu=accountancy&leftmenu=accountancy_accountancy` recognize the auxiliary account

The variable in `htdocs/accountancy/journal/bankjournal.php?mainmenu=accountancy&leftmenu=accountancy_journal&id_journal=3` supposed to carry the auxiliary account is broken.

**Comment by AurelienBISOTTI** [comment]
Okay, I have been able to reproduce the problem

![Capture d’écran du 2023-06-10 21-55-38](https://github.com/Dolibarr/dolibarr/assets/12530585/8bcc9baa-4c9e-47ee-aa36-21f7af370ff9)
![Capture d’écran du 2023-06-10 21-55-57](https://github.com/Dolibarr/dolibarr/assets/12530585/b13b4d07-d2c9-4365-a074-6c0ac4e71e0b)
![Capture d’écran du 2023-06-10 21-56-16](https://github.com/Dolibarr/dolibarr/assets/12530585/c6909f7f-b752-40aa-9c24-a41a84626ad2)

**Comment by ksar-ksar** [comment]
Hello,

Is it still the case on V18 ?

**Comment by thibdrev** [comment]
Hi @ksar-ksar,

Yes, I still observe this behaviour on a V18.0.2.

If I create an expense report, validate it and bound it to a finance account, then I see it normally in the Expense reports Journal:
![image](https://github.com/Dolibarr/dolibarr/assets/7697736/b02d0fc7-ec74-4388-ae87-bc372002c586)

However, once I pay it and check the Bank journal, I still get the message:
![image](https://github.com/Dolibarr/dolibarr/assets/7697736/2add2425-99b4-49ed-b880-328b3973c2af)

It seems that the subledger account is not retrieved.

I have a similar message on a salary payment :
![image](https://github.com/Dolibarr/dolibarr/assets/7697736/174ca2b6-8d4a-4d20-ac10-10843e0cc3f4)

An here as well, the bank entry is rightly linked to a user with an accountancy code (421xxx).

**Comment by StephaneBRT** [comment]
Hello @thibdrev  ,

I confirm. I have the same problem in the bank journal for expense reports on a version 18.0.4.

**Comment by soerpoers** [comment]
same for me on 18.0.4
